src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -55,12 +55,16 @@ public function __construct(string $host = 'localhost', int $port = 0, bool $tls
         if (!$tls) {
             $stream->disableTls();
         } elseif (!$verifyPeer) {
-            $stream->setStreamOptions(array_merge_recursive($stream->getStreamOptions(), [
-                'ssl' => [
-                    'verify_peer' => false,
-                    'verify_peer_name' => false,
-                ],
-            ]));
+            $streamOptions = $stream->getStreamOptions();
+            $sslOptions = [];
+            if (\array_key_exists('ssl', $streamOptions)) {
+                $sslOptions = $streamOptions['ssl'];
+            }
+            $sslOptions['verify_peer'] = false;
+            $sslOptions['verify_peer_name'] = false;
+            $streamOptions['ssl'] = $sslOptions;
```

Comment:
you could simplify all this code by relying on the fact that PHP creates nested arrays for you on write:

```php
$streamOptions = $stream->getStreamOptions();

$streamOptions['ssl']['verify_peer'] = false;
$streamOptions['ssl']['verify_peer_name'] = false;

$stream->setStreamOptions($streamOptions);
```

Reply:
Right! And we could even further reduce the impact by moving this logic to `EsmtpTransportFactory`, that'd save us from adding a new constructor arg.